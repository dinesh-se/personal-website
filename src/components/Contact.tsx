import { SocialIcon } from 'react-social-icons/component';
import 'react-social-icons/email';
import 'react-social-icons/github';
import 'react-social-icons/linkedin';

import { Contact as ContactType } from '@types';

const Contact = ({ linkedin, github, eMail }: ContactType) => {
	return (
		<section>
			<SocialIcon
				url={linkedin}
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
			<SocialIcon
				url={github}
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
			<SocialIcon
				url={`mailto:${eMail}`}
				network="email"
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
		</section>
	);
};

export default Contact;
