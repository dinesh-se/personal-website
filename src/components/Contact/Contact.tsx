import { SocialIcon } from 'react-social-icons';

import { Contact as ContactType } from '@types';

const Contact = ({ linkedin, github, email }: ContactType) => {
	return (
		<section>
			<SocialIcon
				url={linkedin}
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
			<SocialIcon
				url={github}
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
			<SocialIcon
				url={`mailto:${email}`}
				network="email"
				className="social mx-1"
				fgColor="currentColor"
				bgColor="transparent"
			/>
		</section>
	);
};

export default Contact;
