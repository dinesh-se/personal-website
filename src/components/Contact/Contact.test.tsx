import { render } from '@testing-library/react';

import { Contact as ContactType } from '@types';

import Contact from './Contact';

describe('Contact', () => {
	it('renders correctly', () => {
		const props: ContactType = {
			linkedin: 'https://linkedin.com',
			github: 'https://github.com',
			eMail: 'test@test.com',
		};
		const { container } = render(<Contact {...props} />);
		expect(container).toMatchSnapshot();
	});
});
