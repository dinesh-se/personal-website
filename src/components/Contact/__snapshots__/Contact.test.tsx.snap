// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<div>
  <section>
    <a
      aria-label="linkedin"
      class="social-icon social mx-1"
      href="https://linkedin.com"
      style="display: inline-block; width: 50px; height: 50px; position: relative; overflow: hidden; vertical-align: middle;"
    >
      <span
        class="social-container"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
      >
        <svg
          aria-label="linkedin social icon"
          class="social-svg"
          role="img"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
          viewBox="0 0 64 64"
        >
          <g
            class="social-svg-icon"
            style="transition: fill 170ms ease-in-out; fill: currentColor;"
          >
            <path
              d="M0,0H64V64H0ZM0 0v64h64V0zm25.8 44h-5.4V26.6h5.4zm-2.7-19.7c-1.7 0-3.1-1.4-3.1-3.1s1.4-3.1 3.1-3.1 3.1 1.4 3.1 3.1-1.4 3.1-3.1 3.1M46 44h-5.4v-8.4c0-2 0-4.6-2.8-4.6s-3.2 2.2-3.2 4.5V44h-5.4V26.6h5.2V29h.1c.7-1.4 2.5-2.8 5.1-2.8 5.5 0 6.5 3.6 6.5 8.3V44z"
            />
          </g>
          <g
            class="social-svg-mask"
            style="transition: fill 170ms ease-in-out; fill: transparent;"
          >
            <path
              d="M0 0v64h64V0zm25.8 44h-5.4V26.6h5.4zm-2.7-19.7c-1.7 0-3.1-1.4-3.1-3.1s1.4-3.1 3.1-3.1 3.1 1.4 3.1 3.1-1.4 3.1-3.1 3.1M46 44h-5.4v-8.4c0-2 0-4.6-2.8-4.6s-3.2 2.2-3.2 4.5V44h-5.4V26.6h5.2V29h.1c.7-1.4 2.5-2.8 5.1-2.8 5.5 0 6.5 3.6 6.5 8.3V44z"
            />
          </g>
        </svg>
      </span>
    </a>
    <a
      aria-label="github"
      class="social-icon social mx-1"
      href="https://github.com"
      style="display: inline-block; width: 50px; height: 50px; position: relative; overflow: hidden; vertical-align: middle;"
    >
      <span
        class="social-container"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
      >
        <svg
          aria-label="github social icon"
          class="social-svg"
          role="img"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
          viewBox="0 0 64 64"
        >
          <g
            class="social-svg-icon"
            style="transition: fill 170ms ease-in-out; fill: currentColor;"
          >
            <path
              d="M0,0H64V64H0ZM0 0v64h64V0zm37.1 47.2c-.8.2-1.1-.3-1.1-.8V42c0-1.5-.5-2.5-1.1-3 3.6-.4 7.3-1.7 7.3-7.9 0-1.7-.6-3.2-1.6-4.3.2-.4.7-2-.2-4.2 0 0-1.3-.4-4.4 1.6-1.3-.4-2.6-.5-4-.5s-2.7.2-4 .5c-3.1-2.1-4.4-1.6-4.4-1.6-.9 2.2-.3 3.8-.2 4.2-1 1.1-1.6 2.5-1.6 4.3 0 6.1 3.7 7.5 7.3 7.9-.5.4-.9 1.1-1 2.1-.9.4-3.2 1.1-4.7-1.3 0 0-.8-1.5-2.5-1.6 0 0-1.6 0-.1 1 0 0 1 .5 1.8 2.3 0 0 .9 3.1 5.4 2.1v2.7c0 .4-.3.9-1.1.8-6.3-2-10.9-8-10.9-15.1 0-8.8 7.2-16 16-16s16 7.2 16 16c0 7.1-4.6 13.1-10.9 15.2"
            />
          </g>
          <g
            class="social-svg-mask"
            style="transition: fill 170ms ease-in-out; fill: transparent;"
          >
            <path
              d="M0 0v64h64V0zm37.1 47.2c-.8.2-1.1-.3-1.1-.8V42c0-1.5-.5-2.5-1.1-3 3.6-.4 7.3-1.7 7.3-7.9 0-1.7-.6-3.2-1.6-4.3.2-.4.7-2-.2-4.2 0 0-1.3-.4-4.4 1.6-1.3-.4-2.6-.5-4-.5s-2.7.2-4 .5c-3.1-2.1-4.4-1.6-4.4-1.6-.9 2.2-.3 3.8-.2 4.2-1 1.1-1.6 2.5-1.6 4.3 0 6.1 3.7 7.5 7.3 7.9-.5.4-.9 1.1-1 2.1-.9.4-3.2 1.1-4.7-1.3 0 0-.8-1.5-2.5-1.6 0 0-1.6 0-.1 1 0 0 1 .5 1.8 2.3 0 0 .9 3.1 5.4 2.1v2.7c0 .4-.3.9-1.1.8-6.3-2-10.9-8-10.9-15.1 0-8.8 7.2-16 16-16s16 7.2 16 16c0 7.1-4.6 13.1-10.9 15.2"
            />
          </g>
        </svg>
      </span>
    </a>
    <a
      aria-label="email"
      class="social-icon social mx-1"
      href="mailto:test@test.com"
      style="display: inline-block; width: 50px; height: 50px; position: relative; overflow: hidden; vertical-align: middle;"
    >
      <span
        class="social-container"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
      >
        <svg
          aria-label="email social icon"
          class="social-svg"
          role="img"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; border-radius: 50%; fill-rule: evenodd;"
          viewBox="0 0 64 64"
        >
          <g
            class="social-svg-icon"
            style="transition: fill 170ms ease-in-out; fill: currentColor;"
          >
            <path
              d="M0,0H64V64H0ZM41.1 25H22.9l9.1 7.1zm2.9 1.6-12 9.3-12-9.3V39h24zM0 0v64h64V0zm47 42H17V22h30z"
            />
          </g>
          <g
            class="social-svg-mask"
            style="transition: fill 170ms ease-in-out; fill: transparent;"
          >
            <path
              d="M41.1 25H22.9l9.1 7.1zm2.9 1.6-12 9.3-12-9.3V39h24zM0 0v64h64V0zm47 42H17V22h30z"
            />
          </g>
        </svg>
      </span>
    </a>
  </section>
</div>
`;
